### ISSUE TITLE
Negative trees to allow tree support density to be zero

### ISSUE THREAD

**Issue by Hello1024** [ThomasRahm/CuraEngine#4]
So...    These tree supports work really well in most cases with a support infill density of 0%.   That makes intuitive sense - if you want to hold something up in the air, a hollow cone is the way to do it with minimum material usage.

Buuuut...    There is one failure case - and that's when three tree supports merge.   In that case, part of the support tree is unprintable, since the branching point of the tree involves printing in the middle of the 'trunk', which is unprintable.

However...   there is a solution...      Generate tree supports as normal...   Then detect whenever this case happens (ie. a polygon would be drawn on a layer with nothing below it).  Then generate a *new* tree, with the detected point as the tip of one branch.    We shall call this the *negative tree*.  

Within each layer, the printed polygons are then the polygons of the positive tree, *minus* the polygons of the negative tree.

Branches of the negative tree can merge with one another, or they can hit the build plate, or they can travel entirely outside the regular tree (and are then irrelevant).

This approach is recursive...   If, within the negative tree there are three branches meeting, it also leaves an unsupported polygon - so there would be a positive tree within the negative tree, etc.

Thoughts?   (also posted [here](https://www.reddit.com/r/3Dprinting/comments/p7n4x4/i_created_a_new_tree_support_implementation_for/ii0vfmv/) )

**Comment by ThomasRahm** [comment]
Some lines being printed mid-air is an issue, but one I have ignored for the following reasons:
1. A feature to support floating lines should also be active at low infill rates, but infill lines make this a bit harder as wall lines of the inner polygon should not overlap with infill lines.
2. The lines that are mid-air are all inside a branch. It should behave similar to spaghetti infill, and not cause any problems. If I had a way to easily remove just the lines mid-air I would have done it, as I don't think that the absence of these lines causes stability issues.
3. When connecting an area inside the branch to the outer wall of the branch I have two choices: I either place the lines next to the outer wall, and hope it sticks and Cura chooses to print the outer support wall before the "inner" one or i make the lines overlap, which could cause print failures similar to over-extrusion. I don't like any of these two choices. A proper solution would need access to how the lines are generated, but support generation just creates areas that are then filled with infill and the correct amount of walls.

**If you encounter any print failures because of these lines, please send me a picture and a description of how it failed and I will definitely reconsider adding something to support or remove these lines.** 

In short I do think it looks ugly, but shouldn't cause issues. And I don't care if the support looks ugly, but just that it works well.
